Datastore/src/EntityTrait.php

```diff
@@ -0,0 +1,156 @@
+     *
+     *     @type string $cursor Set only when the entity is obtained by a query
+     *           result. If set, the entity cursor can be retrieved from
+     *           {@see Google\Cloud\Datastore\Entity::cursor()}.
+     *     @type string $baseVersion Set only when the entity is obtained by a
+     *           query result. If set, the entity cursor can be retrieved from
+     *           {@see Google\Cloud\Datastore\Entity::baseVersion()}.
+     *     @type array $excludeFromIndexes A list of entity keys to exclude from
+     *           datastore indexes.
+     *     @type array $meanings A list of meaning values for entity properties.
+     *     @type bool $populatedByService Indicates whether the entity was
+     *           created as the result of a service request.
+     * }
+     */
+    public function __construct(Key $key = null, array $entity = [], array $options = [])
+    {
+        $this->key = $key;
+        $this->entity = $entity;
+        $this->options = $options + [
+            'cursor' => null,
+            'baseVersion' => null,
+            'populatedByService' => false,
+            'excludeFromIndexes' => [],
+            'meanings' => []
+        ];
+    }
+
+    /**
+     * A factory method, used by the Datastore client to create entities.
+     *
+     * @param Key $key The Entity's Key, defining its unique identifier.
+     * @param array $entity [optional] The entity body.
+     * @param array $options [optional] {
+     *     Configuration Options
+     *
+     *     @type string $cursor Set only when the entity is obtained by a query
+     *           result. If set, the entity cursor can be retrieved from
+     *           {@see Google\Cloud\Datastore\Entity::cursor()}.
+     *     @type string $baseVersion Set only when the entity is obtained by a
+     *           query result. If set, the entity cursor can be retrieved from
+     *           {@see Google\Cloud\Datastore\Entity::baseVersion()}.
+     *     @type array $excludeFromIndexes A list of entity keys to exclude from
+     *           datastore indexes.
+     *     @type array $meanings A list of meaning values for entity properties.
+     *     @type bool $populatedByService Indicates whether the entity was
+     *           created as the result of a service request.
+     * }
+     * @throws InvalidArgumentException
+     */
+    public static function factory(Key $key = null, array $entity = [], array $options = [])
+    {
+        return new static($key, $entity, $options);
+    }
+
+    /**
+     * Get the entity data
+     *
+     * Example:
+     * ```
+     * $data = $entity->get();
+     * ```
+     *
+     * @return array
+     */
+    public function get()
+    {
+        return $this->entity;
+    }
+
+    /**
+     * Set the entity data
+     *
+     * Calling this method replaces the entire entity body. To add or modify a
+     * single value on the entity, use the array syntax for assignment.
```

Comment:
Given that users defining their own implementation of `EntityInterface` may choose to not also implement array access, should we provide a simple helper on this trait that will allow them to set single values as well?

Reply:
https://github.com/GoogleCloudPlatform/google-cloud-php/pull/913/commits/29421a5d72ac6ab2032d93251a07a28e30a99bc4#diff-ef2b092e0f5323a15a9e2414f7e83532R128